### DISCUSSION TITLE
[Bug]: Getting ERR_EMPTY_RESPONSE trying to run ngrok server on Termux

### DISCUSSION THREAD

**Discussion by artisticfox8** [termux/termux-packages#7485] [Q&A] [answered]
### Problem description

I want to run a Python server on Android using Python FastAPI package. For that I need a server which will run my program, such as daphne. I also need to make it online, with the help of a service like pagekite or ngrok. As Pagekite had some server issues and it's servers were unreachable, I've decided to use ngrok.
To make it work, I used this tutorial https://wiki.termux.com/wiki/Bypassing_NAT#Ngrok
I followed the steps on using ngrok over ssh carefully.

### What steps will reproduce the bug?

I followed the steps on using ngrok over ssh carefully. When I run `ssh -R 0:localhost:8000 tunnel.us.ngrok.com tcp 8000` and navigate to the adress returned (replacing tcp with http), I get ERR_EMPTY_RESPONSE in Chrome.
My server works, when I go to 0.0.0.0:8000 I get a JSON response.

### What is the expected behavior?

I get the JSON response when going to the ngrok provided online public IP.

### System information

termux-info:

```
Will add in a few hours, I don't have the device with me right now, sorry :)

```

**Comment by ghost** [comment, marked as answer]
Switch to http mode
```
ssh -R 0:localhost:8000 tunnel.us.ngrok.com http 8000
```

Try to use curl instead, especially if server returns a plain json output:
```
curl http://ngrokurl:12345
```

***

Just have tried it with `python -m http.server 8000` - it worked in both plain tcp and http forwarding. So issue is not with Ngrok, instructions provided on Wiki or Termux itself.

**Reply by artisticfox8** [reply]
Thanks! The issue was with you told me on reddit I can run the command with tcp, replace tcp with http, which isn't the case. With this command where the word tcp is replaced with  http when I visit the adress, it works.
